iocage/lib/Jail.py

```diff
@@ -218,6 +232,22 @@ def __init__(
                 "id": self._resolve_name(data)
             }
 
+        JailResource.__init__(
+            self,
+            jail=self,
+            host=self.host,
+            logger=self.logger,
+            zfs=self.zfs,
+            **resource_args
+        )
+
+        if not new and (("id" not in data) or (data["id"] is None)):
```

Comment:
this looks like a candidate for encapsulation ;)

Reply:
It's below 80 characters!!!1 😼

Reply:
yea, but, the logic is too much for my tired brain 😩